internal/driver/driver.go

```diff
@@ -329,12 +361,46 @@ func (d *Driver) HandleWriteCommands(deviceName string, protocols map[string]mod
 	}
 
 	for i, req := range reqs {
-		command, ok := req.Attributes[Command]
+		command, ok := req.Attributes[SetFunction]
 		if !ok {
 			return errors.NewCommonEdgeX(errors.KindContractInvalid,
 				fmt.Sprintf("command for USB camera resource %s is not specified, please check device profile",
 					req.DeviceResourceName), nil)
 		}
+		queryParams, edgexErr := getQueryParameters(req)
+		if edgexErr != nil {
+			return errors.NewCommonEdgeXWrapper(edgexErr)
+		}
+
+		// flush out the query so it resets with new calls
+		if _, ok := req.Attributes[UrlRawQuery]; ok {
+			req.Attributes[UrlRawQuery] = ""
+		}
+
+		var videoPath string
+		pathIndex := queryParams.Get(PathIndex)
+		if len(pathIndex) == 0 {
+			// currently defaults to using the first available stream
+			videoPath = device.paths[0]
+		} else {
+			pathIndexConv, err := strconv.Atoi(pathIndex)
+			if err != nil {
+				return err
+			}
+			if pathIndexConv >= len(device.paths) {
+				return errors.NewCommonEdgeX(errors.KindIOError,
+					fmt.Sprintf("Video streaming path does not exist for the device %v at PathIndex %d", device.name, pathIndexConv), nil)
+			}
+			videoPath = device.paths[pathIndexConv]
+		}
+
+		cameraDevice, err := usbDevice.Open(videoPath)
+		if err != nil {
+			return errors.NewCommonEdgeX(errors.KindServerError,
+				fmt.Sprintf("failed to open the underlying device at specified path %s", videoPath), err)
+		}
+		defer cameraDevice.Close()
```

Comment:
same for this too as above.

Reply:
Also this whole code of reading the request, flushing out the query till opening the device for streaming can be made into a separate function as this is repeated 2 times in the same file.